WIP: Welcome Mexican documentation v13.0 (#605)
odoo/documentation-user: accounting/localizations/mexico.rst

Configuration
=============

Enable electronic invoicing
---------------------------

From the home screen, go to :menuselection:`Settings --> General Options`. In the list of applications on the left, tap on 
Accounting. On the resulting page, look for the Invoices section and activate the “Mexican Electronic Invoicing” option. Finally, press the 
button that says Save, top left. With this you can generate the signed invoice and also generate the signed payment complement, all 
automatically integrated into the normal billing flow in Odoo.

.. image:: media/mexico53.png
   :align: center

Company Information
-------------------

First, make sure that your company is configured with the correct data. Go to :menuselection:`Settings --> Companies --> Update Info` and enter a 
valid address and VAT for your company.

.. image:: media/mexico54.png
   :align: center

If you want use the Mexican localization on test mode, you can put any known address inside Mexico with all fields for the company address 
and set the vat to **EKU9003173C9** for testing purposes or set your own, if you loaded demo data this will be done automatically.

.. image:: media/mexico03.png
   :align: center

> In the form for the company set the proper information for your mexican company (Remember you must to set Mexico as country 

Reviewer: ```suggestion
Set the proper information for your Mexican company in the company information form. (Remember you must to set Mexico as the country 
```

> in your company in order to allow all to work properly following mexican workflow), 

Reviewer: ```suggestion
in your company in order to allow everything to work properly with regards to Mexican workflows).
```

> **Please be sure to not add a new country Mexico by mistake, select the one from the list which is pre created**.

Reviewer: ```suggestion
**Please make sure you do NOT create a new country called "Mexico" by mistake; always select the one from the list of pre-defined countries**.
```

.. image:: media/mexico55.png
   :align: center

Set the proper Fiscal Regime
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Go in :menuselection:`Accounting --> Settings` and enter a valid Fiscal Regime.

Reviewer: ```suggestion
Go to :menuselection:`Accounting --> Settings` and enter a valid Fiscal Regime.
```

.. image:: media/mexico04.png
   :align: center

.. tip::

>    For testing environment you must set **General de Ley Personas Morales**

Reviewer: ```suggestion
   In testing environments you must set **General de Ley Personas Morales**
```

a. From the invoice document.
b. Payment first and then reconcile with invoices.
c. From the bank statements (Once you reconcile a bank statement line and this does not have a payment yet done from a or b).

**a.** From the invoice document.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Create a payment from the invoices that are related, either going to the form view and clicking the **Register Payment** button and from 
the wizard in tha list view of invoices.

.. image:: media/mexico58.png
   :align: center

.. image:: media/mexico59.png
   :align: center

It will open the wizard to set the proper values, in case of Mexico it is important you set the proper **Payment Way** you are 
receiving the payment.

.. image:: media/mexico60.png
   :align: center

Your payment will be automatically signed (you can look for the payment going to the invoice and opening the payment related) if the invoice(s)
are PUE the signature will not be done but it can be forced from the payment itself.

.. image:: media/mexico61.png
   :align: center

.. image:: media/mexico62.png
   :align: center

**b.** Payment first and then reconcile with invoices.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> TODO: CONTINUE THIS

Reviewer 2: @nhomar I think that we should only specify that this option uses the standard process.

**Legal considerations**

- A cancelled invoice will automatically cancelled on the SAT.
- If you retry to use the same invoice after cancelled, you will have as much cancelled CFDI as you tried, then all
  those xml are important to maintain a good control of the cancellation reasons.
- You must unlink all related payment done to an invoice on odoo before cancel such document, this payments must be
  cancelled to following the same approach but setting the "Allow Cancel Entries" in the payment itself.

Credit Notes
------------

> # todo

Reviewer 2: @nhomar we can remove this todo an  add it on new PR once the documentation for Credit Notes is ready.

Debit Notes
------------

> # todo

Reviewer 2: @nhomar we can remove this todo an add it on new PR once the documentation for Debit s is ready.

Payment in advance
-------------------

> # todo

Reviewer 2: @nhomar we can remove this todo an add it on new PR once the documentation for Advance Payments is ready
